src/reconnect-strategy/mqtts.reconnect-strategy.default.ts

```diff
@@ -0,0 +1,40 @@
+import { MqttsReconnectStrategy } from './mqtts.reconnect-strategy';
+import { ConnectError } from '../errors';
+import {promisify} from 'util';
+const sleep = promisify(setTimeout);
+
+export class MqttsReconnectStrategyDefault implements MqttsReconnectStrategy {
+    /**
+     * Internal attempts counter
+     */
+    #attempts = 1;
+    constructor(
+        /**
+         * Maximum attempts amount
+         */
+        private attempts: number = 60,
```

Comment:
This should have a better name. It conflicts with `#attempts`.

```suggestion
        private maxAttempts: number = 60,
```